static/src/javascripts/projects/common/modules/preferences/main.js

```diff
@@ -73,8 +75,122 @@ define([
+                            });
+                    });
+                };
+
+                var unsubscribe = function () {
+                    // Disable the button so it can't be changed while
+                    // we process the permission request
+                    pushButton.disabled = true;
+
+                    return navigator.serviceWorker.ready.then(function (serviceWorkerRegistration) {
+                        return serviceWorkerRegistration.pushManager.getSubscription()
+                            .then(function (subscription) {
+                                if (!subscription) {
+                                    // No subscription? Reset UI
+                                    pushButton.disabled = false;
+                                    updateState({ pushEnabled: false });
+                                } else {
+                                    return subscription.unsubscribe()
+                                        .then(function () {
+                                            pushButton.disabled = false;
+                                            updateState({ pushEnabled: false });
+
+                                            return sendSubscription(subscription, { delete: true });
+                                        });
+                                }
+                            });
+                    });
+                };
+
+                var sendSubscription = function (subscription, options) {
+                    options = options || {};
+                    // TODO: var url
+
+                    var mobileNotificationsWebHost = config.page.pushNotificationsHost;
+
+                    if (!mobileNotificationsWebHost) {
+                        throw new Error('No notifications host found');
+                    }
+
+                    /*global fetch, Headers*/
+                    return fetch(mobileNotificationsWebHost + '/web/subscription', {
+                        method: options.delete ? 'DELETE' : 'POST',
+                        headers: new Headers({ 'Content-Type': 'application/json' }),
+                        // TODO: Support deprecated subscriptionId (now part of endpoint)
+                        body: JSON.stringify({
+                            subId: subscription.subscriptionId,
+                            endpoint: subscription.endpoint.replace('/' + subscription.subscriptionId, '')
+                            // TODO: edition based notifications
+                            //edition: config.page.edition
+                        })
+                    });
+                };
+
+                pushButton.addEventListener('click', function () {
+                    if (isPushEnabled) {
+                        unsubscribe();
+                    } else {
+                        subscribe();
+                    }
+                });
+
+                navigator.serviceWorker.register('/2015-05-19-service-worker.js')
```

Comment:
So the plan is to switch it on on CODE? It'll still break that environment for all other people using it. Isn't it better to put it behind a 0% A/B test?